### PR TITLE
[14.0][MIG] sale_coupon_order_line_link: Migration to 14.0

### PR DESCRIPTION
**Pull request by ilyasProgrammer**

### CODE REVIEW

File: sale_coupon_order_line_link/__manifest__.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
     "name": "Link coupons to order lines",
     "summary": "Adds a link between coupons and their generated order lines for easing "
     "tracking",
-    "version": "13.0.1.2.0",
+    "version": "14.0.1.0.0",
     "development_status": "Production/Stable",
     "category": "Sale",
     "website": "https://github.com/OCA/sale-promotion",
-    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "author": "Ilyas, Ooops404, Tecnativa, Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline sale_coupon_order_line_link/__manifest__.py]: I'm afraid a migration of this scope doesn't grant you co-authorship. You can put yourself on contributors though

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Is any migration script needed?

**Comment by ilyasProgrammer** [comment]
> Is any migration script needed?

As i understand it is not required here. But i'm not 100% sure.

**Comment by ivs-cetmix** [comment]
> 

The migration script is needed if the data structure was changed in the new version. E.g. in v14 account.move field "type" was renamed into "move_type". If there are any similar cases in this module then the migration scripts are mandatory

**Comment by elvise** [comment]
@pedrobaeza is it good now?  :)

**Review by francesco-ooops — APPROVED** [review]
Functional review ok!

**Review by SicurSam — APPROVED** [review]
Functional review ok!

**Comment by elvise** [comment]
@pedrobaeza any news for this ?

**Comment by elvise** [comment]
@OCA/crm-sales-marketing-maintainers @pedrobaeza there is someone for review this PR ?

File: sale_coupon_order_line_link/__manifest__.py (hunk 1 above)
Comment (elvise) [reply sale_coupon_order_line_link/__manifest__.py]: @pedrobaeza is it ok now ?

**Comment by rousseldenis** [comment]
@ilyasProgrammer Thanks for this. Could you squash your last two commits into one ?

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by francesco-ooops** [comment]
@rousseldenis can we also merge please? 🥵

**Comment by rousseldenis** [comment]
> @rousseldenis can we also merge please? hot_face

The merge is conditioned to approvals. There is still a pending change request from @pedrobaeza 

You should always first ping those users :

image

**Comment by francesco-ooops** [comment]
> > @rousseldenis can we also merge please? hot_face
> 
> The merge is conditioned to approvals. There is still a pending change request from @pedrobaeza
> 
> You should always first ping those users :
> 
> image

"Formally" I understand, but the remarks by @pedrobaeza were addressed a [week ago](https://github.com/OCA/sale-promotion/commit/7879de5b5e6021234db4ac585b35a367916c391a ), so that should not prevent merge imho

anyway, @pedrobaeza could you update your review please?

**Comment by pedrobaeza** [comment]
I'm all week on the Spanish Odoo days with little time to other activities. I'll review ASAP.

**Comment by francesco-ooops** [comment]
> I'm all week on the Spanish Odoo days with little time to other activities. I'll review ASAP.

thanks!

**Review by pedrobaeza — APPROVED** [review]
This will require migration script, but that's something out of the scope of this PR

**Pull request merged**